core/src/main/scala/ai/lum/odinson/ExtractorEngine.scala

```diff
@@ -150,6 +152,47 @@ class ExtractorEngine(
     TokenStreamUtils.getTokens(docID, fieldName, indexSearcher, analyzer)
   }
 
+  def query(extractionQuery: ExtractionQuery): OdinResults = {
+
+    // combine the odinson query if needed
+    val odinsonQuery = (extractionQuery.odinsonQuery, extractionQuery.documentLuceneQuery) match {
+      case (None, None) => None
+      case (Some(oq), Some(dq)) => Some(compiler.mkQuery(oq, dq))
+      case (Some(oq), None) => Some(compiler.mkQuery(oq))
+      case _ => throw new IllegalArgumentException("Document query can not be defined without a corresponding odinson query")
+    }
+
+    // parse sentence query
+    val sentenceQuery = extractionQuery.sentenceLuceneQuery.map { sq =>
+      val queryParser = new QueryParser(
+        "word", // default to searching over words
+        analyzer)
+
+      queryParser.parse(sq)
+    }
+
+    // final combine of all query parts to create a single lucene query
+    val fullQuery = (odinsonQuery, sentenceQuery) match {
+      case (Some(oq), Some(sq)) =>
+        new OdinsonFilteredQuery(oq, sq)
+      case (None, Some(sq)) => sq // TODO: convert to some kind of odinson query
+      case (Some(oq), None) => oq
+      case _ => throw new IllegalArgumentException("Either odinson query or a sentence query have to be defined")
+    }
+
+    val numberOfDocuments = extractionQuery.numDocuments.getOrElse(numDocs())
+
+    // run the query
+    (fullQuery, extractionQuery.continuation) match {
+      case (q: OdinsonQuery, Some(c)) => indexSearcher.odinSearch(c, q, numberOfDocuments)
```

Comment:
call all the options, and check that the response makes sense:
- odinson, odinson+search, search, continuation

Reply:
tests added in `ai.lum.odinson.ExtractorEngineTest`